cmd/cone/task_search.go

```diff
@@ -69,7 +69,7 @@ func searchTasksRun(cmd *cobra.Command, args []string) error {
 	case "certify":
 		taskType.Certify = &shared.TaskTypeCertify{}
 	}
-	if len(taskTypes) != 0 {
+	if taskType != (shared.TaskType{}) {
```

Comment:
Hmm, I don't really like comparing to an empty struct like this. Seems like it could cause headaches down the line. What do you think about either of these options? 

```golang
var taskTypes []shared.TaskType
var taskType *shared.TaskType
switch strings.ToLower(v.GetString(taskTypeFlag)) {
	case "grant":
		taskType = &shared.TaskType{Grant: &shared.TaskTypeGrant{}}
	case "revoke":
                taskType = &shared.TaskType{Revoke: &shared.TaskTypeRevoke{}}
        case "certify":
		taskType = &shared.TaskType{Certify: &shared.TaskTypeCertify{}}
}

if taskType != nil {
  taskTypes = []shared.TaskType{*taskType}
}
```

or

```golang
var taskTypes []shared.TaskType
switch strings.ToLower(v.GetString(taskTypeFlag)) {
	case "grant":
		taskTypes = append(taskTypes, shared.TaskType{Grant: &shared.TaskTypeGrant{}})
	case "revoke":
		taskTypes = append(taskTypes, shared.TaskType{Revoke: &shared.TaskTypeRevoke{}})
        case "certify":
		taskTypes = append(taskTypes, shared.TaskType{Certify: &shared.TaskTypeCertify{}})
}
```